datafusion/core/src/physical_plan/windows/mod.rs

```diff
@@ -262,7 +262,22 @@ pub(crate) fn window_ordering_equivalence(
             .iter()
             .cloned(),
     );
-    let out_ordering = input.output_ordering().unwrap_or(&[]);
+    let mut normalized_out_ordering = vec![];
+    for item in input.output_ordering().unwrap_or(&[]) {
+        // To account for ordering equivalences, first normalize the expression:
+        let normalized = normalize_expr_with_equivalence_properties(
+            item.expr.clone(),
+            input.equivalence_properties().classes(),
+        );
+        // Currently we only support, ordering equivalences for `Column` expressions.
+        // TODO: Add support for ordering equivalence for all `PhysicalExpr`s
```

Comment:
I agree this would be nice to do.

I suggest if you want to head in this direction, we start trying to encapsulate the code for working with equivalence classes, required sort order, and input orderings into a single location that is disconnected from the individual physical operators

Like I wonder if we could end up with an API like

```rust
let mut builder = OrderingEquivalenceBuilder::new(input.output_ordering())
  .with_equivalences(input.equivalence_properties());

...

            // Only the built-in `RowNumber` window function introduces a new
            // ordering:
            if builtin_window_expr
                .get_built_in_func_expr()
                .as_any()
                .is::<RowNumber>()
            {
                        let column = Column::new(field.name(), idx);
                        let options = SortOptions {
                            descending: false,
                            nulls_first: false,
                        }; // ASC, NULLS LAST
                        let rhs = OrderedColumn::new(column, options);

               builder.add_equal_condition(rhs)
          }
...
builder.build()
...